File: JS/Tema3_Ejercicio7.js
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 7
 * ================================================
 * Determinar si un carácter es vocal o consonante.
 * 
 * EXPLICACIÓN:
 * - Métodos de string: toLowerCase(), includes()
 * - Array de vocales para verificación
 * - Validación que sea una letra
 */

function clasificarCaracter(caracter) {
    if (caracter.length !== 1) {
        return "Ingrese un solo carácter";
    }
    
    const vocal = caracter.toLowerCase();
    const vocales = ['a', 'e', 'i', 'o', 'u'];
    
    if (vocales.includes(vocal)) {
        return `'${caracter}' es una VOCAL`;
    } else if (/[a-z]/i.test(caracter)) {
        return `'${caracter}' es una CONSONANTE`;
    } else {
        return `'${caracter}' no es una letra`;
    }
}

console.log("=== VOCAL O CONSONANTE ===");
const caracteres = ['A', 'b', 'E', 'x', '3', 'o'];

caracteres.forEach(c => {
    console.log(clasificarCaracter(c));
});

/*
 * SALIDA ESPERADA:
 * === VOCAL O CONSONANTE ===
 * 'A' es una VOCAL
 * 'b' es una CONSONANTE
 * 'E' es una VOCAL
 * 'x' es una CONSONANTE
 * '3' no es una letra
 * 'o' es una VOCAL
 */

File: Java/Tema3_Ejercicio7.java
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 7
 * =================================================
 * Escriba un programa que determine si un carácter es una vocal o consonante.
 */

import java.util.Scanner;

public class Tema3_Ejercicio7 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite un carácter: ");
        String input = scanner.nextLine().toLowerCase();
        
        System.out.println("\n=== CLASIFICACIÓN DE CARÁCTER ===");
        
        if (input.length() != 1) {
            System.out.println("Por favor ingrese solo un carácter.");
        } else {
            char c = input.charAt(0);
            
            if (!Character.isLetter(c)) {
                System.out.println("'" + c + "' NO es una letra.");
            } else if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') {
                System.out.println("'" + c + "' es una VOCAL.");
            } else {
                System.out.println("'" + c + "' es una CONSONANTE.");
            }
        }
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite un carácter: a
 * 
 * === CLASIFICACIÓN DE CARÁCTER ===
 * 'a' es una VOCAL.
 */
